lib/diplomat/check.rb

```diff
@@ -54,28 +54,41 @@ def deregister(check_id)
       ret.status == 200
     end
 
-    # Pass a check
+    # Update a TTL check
     # @param check_id [String] the unique id of the check
+    # @param status [String] status of the check. Valid values are "passing", "warning", and "critical"
+    # @param output [String] human-readable message will be passed through to the check's Output field
     # @return [Integer] Status code
-    def pass(check_id)
-      ret = @conn.put "/v1/agent/check/pass/#{check_id}"
+    def update_ttl(check_id, status, output = nil)
+      ret = @conn.put do |req|
+        req.url "/v1/agent/check/update/#{check_id}"
+        req.body = JSON.generate('Status' => status, 'Output' => output)
```

Comment:
According to the doc, it should use the query parameter "note", not by using JSON payload. Did you test this?

Reply:
My first production implementation was with "note" query params, but I can't send complex data(json) without URL-encoding. After that I change to generic "/agent/check/update/:check_id" update call and all works well.

There is one issue: Output is silently ignored in "critical" update without any explanation in consul docs. I will open consul issue for this case soon.

Reply:
warn and pass works as expected in docs